otx/cli/manager/config_manager.py

```diff
@@ -0,0 +1,405 @@
+        self.update_data_config(data_yaml)
+
+    def _get_train_type(self) -> str:
+        """Check and return the train_type received as input args."""
+        args_hyper_parameters = gen_params_dict_from_args(self.args)
+        algo_backend = args_hyper_parameters.get("algo_backend", False)
+        if algo_backend:
+            train_type = algo_backend.get("train_type", {"value": "INCREMENTAL"})
+            return train_type.get("value", "INCREMENTAL")
+        return "INCREMENTAL"
+
+    def auto_task_detection(self, data_roots: str) -> str:
+        """Detect task type automatically."""
+        self.data_format = self.dataset_manager.get_data_format(data_roots)
+        print(f"[*] Detected dataset format: {self.data_format}")
+        return self._get_task_type_from_data_format(self.data_format)
+
+    def _get_task_type_from_data_format(self, data_format: str) -> str:
+        """Detect task type.
+
+        For some datasets (i.e. COCO, VOC, MVTec), can't be fully automated.
+        Because those datasets have several format at the same time.
+        (i.e. for the COCO case, object detection and instance segmentation annotations coexist)
+        In this case, the task_type will be selected to default value.
+
+        For action tasks, currently action_classification is default.
+
+        If Datumaro supports the Kinetics, AVA datasets, MVTec, _is_cvat_format(), _is_mvtec_format()
+        functions will be deleted.
+        """
+
+        for task_key, data_value in TASK_TYPE_TO_SUPPORTED_FORMAT.items():
+            if data_format in data_value:
+                self.task_type = task_key
+                print(f"[*] Detected task type: {self.task_type}")
+                return task_key
+        raise ValueError(f"Can't find proper task. we are not support {data_format} format, yet.")
+
+    def auto_split_data(self, data_roots: str, task: str):
+        """Automatically Split train data --> train/val dataset."""
+        self.data_format = self.dataset_manager.get_data_format(data_roots)
+        dataset = self.dataset_manager.import_dataset(data_root=data_roots, data_format=self.data_format)
+        train_dataset = self.dataset_manager.get_train_dataset(dataset)
+        val_dataset = self.dataset_manager.get_val_dataset(dataset)
+        splitted_dataset = None
+        if self.data_format in AUTOSPLIT_SUPPORTED_FORMAT:
+            if val_dataset is None:
+                splitted_dataset = self.dataset_manager.auto_split(
+                    task=task,
+                    dataset=train_dataset,
+                    split_ratio=[("train", 0.8), ("val", 0.2)],
+                )
+            else:
+                print(f"[*] Found validation data in your dataset in {data_roots}. It'll be used as validation data.")
+                splitted_dataset = {"train": train_dataset, "val": val_dataset}
+        else:
+            print(f"[*] Current auto-split can't support the {self.data_format} format.")
```

Comment:
This will be addressed after further discussion. In the current PR, I decided to leave it as it is for now. Most of the case the training doesn't work, or it might work with selfsl.